app/models/validations/sales/soft_validations.rb

```diff
@@ -37,9 +37,29 @@ def deposit_over_soft_max?
     deposit > savings * 4 / 3
   end
 
-  def hodate_3_years_or_more_saledate?
-    return unless hodate && saledate
+  def hodate_3_years_or_more_exdate?
+    return unless hodate && exdate
 
-    ((saledate.to_date - hodate.to_date).to_i / 365) >= 3
+    ((exdate.to_date - hodate.to_date).to_i / 365) >= 3
+  end
+
+  def purchase_price_out_of_soft_range?
+    return unless value && beds && la
+
+    purchase_price_range.present? && !value.between?(purchase_price_range.soft_min, purchase_price_range.soft_max)
+  end
+
+  def purchase_price_min_or_max_text
+    value < purchase_price_range.soft_min ? "minimum" : "maximum"
+  end
+
+  def purchase_price_soft_min_or_soft_max
+    value < purchase_price_range.soft_min ? purchase_price_range.soft_min : purchase_price_range.soft_max
+  end
+
+private
+
+  def purchase_price_range
+    LaPurchasePriceRange.find_by(start_year: collection_start_year, la:, bedrooms: beds)
```

Comment:
should there be a default year for sales before the 2022/3 collection year? Or do we not care about validating old data.

Reply:
I think we're unlikely to have 21/22 data, but we will have to add ranges for 23/24 before April